azure-iot-device/azure/iot/device/iothub/sync_handler_manager.py

```diff
@@ -0,0 +1,232 @@
+
+    @property
+    def on_twin_desired_properties_patch_received(self):
+        return self._on_twin_desired_properties_patch_received
+
+    @on_twin_desired_properties_patch_received.setter
+    def on_twin_desired_properties_patch_received(self, value):
+        self._generic_handler_setter(TWIN_DP_PATCH, value)
+
+
+class SyncHandlerManager(AbstractHandlerManager):
+    """Handler manager for use with synchronous clients"""
+
+    def _inbox_handler_runner(self, inbox, handler_name):
+        """Run infinite loop that waits for an inbox to receive an object from it, then calls
+        the handler with that object
+        """
+        logger.debug("HANDLER RUNNER ({}): Starting runner".format(handler_name))
+        # Define a callback that can handle errors in the ThreadPoolExecutor
+
+        def _handler_callback(future):
+            try:
+                e = future.exception()
+            except concurrent.futures.CancelledError as raised_e:
+                new_err = HandlerManagerException(
+                    message="HANDLER ({}): Invocation unexpectedly ended with cancellation".format(
+                        handler_name
+                    ),
+                    cause=raised_e,
+                )
+                handle_exceptions.handle_background_exception(new_err)
+            else:
+                if e:
+                    new_err = HandlerManagerException(
+                        message="HANDLER ({}): Error during invocation".format(handler_name),
+                        cause=e,
+                    )
+                    handle_exceptions.handle_background_exception(new_err)
+                else:
+                    logger.debug(
+                        "HANDLER ({}): Successfully completed invocation".format(handler_name)
+                    )
+
+        # Run the handler in a threadpool, so that it cannot block other handlers (from a different task),
+        # or the main client thread. The number of worker threads forms an upper bound on how many instances
+        # of the same handler can be running simultaneously.
+        tpe = concurrent.futures.ThreadPoolExecutor(max_workers=4)
+        while True:
+            handler_arg = inbox.get()
+            if isinstance(handler_arg, HandlerRunnerKillerSentinel):
+                # Exit the runner when a HandlerRunnerKillerSentinel is found
+                logger.debug(
+                    "HANDLER RUNNER ({}): HandlerRunnerKillerSentinel found in inbox. Exiting.".format(
+                        handler_name
+                    )
+                )
+                tpe.shutdown()
+                break
+            # NOTE: we MUST use getattr here using the handler name, as opposed to directly passing
+            # the handler in order for the handler to be able to be updated without cancelling
+            # the running task created for this coroutine
+            handler = getattr(self, handler_name)
```

Comment:
this handler is constant for the inbox and name right ? we are only in a loop to handle various items in the same inbox ? so any reason for doing this inside the loop